src/command_modules/azure-cli-storage/azure/cli/command_modules/storage/_validators.py

```diff
@@ -79,7 +78,9 @@ def get_config_value(section, key, default):
         n.account_name = conn_dict.get('AccountName')
         n.account_key = conn_dict.get('AccountKey')
         if not n.account_name or not n.account_key:
-            raise CLIError('Connection-string: %s, is malformed.' % n.connection_string)
+            from knack.util import CLIError
+            raise CLIError('Connection-string: %s, is malformed. Some shell environments require the '
+                           'connection string to be surrounded by quotes.' % n.connection_string)
```

Comment:
For CLIErrors, you can use the .format syntax.  The weird %s syntax is only required for logging messages. :)